libV2/server/API/metrics/ingestMetric.js

```diff
@@ -0,0 +1,20 @@
+const errors = require('../../../errors');
+const { UtapiMetric } = require('../../../models');
+const { client: cacheClient } = require('../../../cache');
+
+async function ingestMetric(ctx, params) {
+    let metrics;
+    try {
+        metrics = params.body.map(m => new UtapiMetric(m));
+    } catch (error) {
+        throw errors.InvalidRequest;
+    }
+    try {
+        await Promise.all(metrics.map(m => cacheClient.pushMetric(m)));
+    } catch (error) {
+        throw errors.ServiceUnavailable;
+    }
```

Comment:
You could combine the catches I think
```suggestion
    try {
        let metrics = params.body.map(m => new UtapiMetric(m));
        await Promise.all(metrics.map(m => cacheClient.pushMetric(m)));
    } catch (error) {
        throw errors.InvalidRequest || errors.ServiceUnavailable;
    }
```

Not sure if this any more readable but maybe.

Reply:
Wouldn't this `throw errors.InvalidRequest || errors.ServiceUnavailable;` always throw `errors.InvalidRequest`?

Reply:
Uh, let me do a test, I thought it would throw Invalid request if it was defined if not Service Unavailable.